concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -1178,7 +1213,16 @@ handleContractUpdateV1 originAddr istance checkAndGetSender transferAmount recei
                 -- Add the transfer to the current changeset and return the corresponding event.
                 lift (withContractToAccountAmountV1 (instanceAddress senderInstance) targetAccount tAmount $
                       return [Transferred addr tAmount (AddressAccount accAddr)])
-
+          handleContractUpgrade :: ContractAddress
+                                -- ^The address of the instance to update.
+                                -> ModuleRef
+                                -- ^The new module that the instance should be using for execution.
+                                -> ExceptT WasmV1.EnvFailure (LocalT r m) [Event] -- ^The events resulting from the transfer.
+                                -- ^The events resulting from the upgrade.
+          handleContractUpgrade cAddr modRef = do
+              -- Add the upgrade to the 'ChangeSet'.
+              lift $! addContractUpgrade cAddr modRef
+              return [Updated cAddr undefined undefined undefined undefined undefined []]
```

Comment:
As discussed in the office, this should be `UPgraded oldModule newModule` instead.